main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -526,8 +526,8 @@ object Weeder {
           @@(elms map visit) map {
             case es =>
               val empty = mkApply("Set/empty", Nil, sp1, sp2)
-              es.foldLeft(empty) {
-                case (acc, elm) => mkApply("Set/insert", List(elm, acc), sp1, sp2)
+              es.foldRight(empty) {
```

Comment:
Great catch! Although the previous implementation is actually correct.

Think of it this way, in the expression #{1, 2, 3} you would expect the elements to be inserted in the order 1, 2, 3 (and thus foldLeft). However, when we use a list to implement as set, you do end up with the some what confusing situation where #{1, 2, 3} is internally represented as Set(List(3, 2, 1)), but that is quiet alright.

Instead, what you probably want, is a function `eq` which compares to sets for equality by testing if every element in one set appears in the other, and vice versa. (Come to think of it, I should probably disallow == on sets, and force you to use this `eq` function).